lib/util/host-rules.spec.ts

```diff
@@ -107,6 +110,68 @@ describe(getName(), () => {
           .token
       ).toBeUndefined();
     });
+    it('matches on specific path', () => {
+      // Initialized platform holst rule
+      add({
+        hostType: PLATFORM_TYPE_GITHUB,
+        matchHost: 'https://api.github.com',
+        token: 'defaultToken',
+      });
+      // Initialized generic host rule for github platform
+      add({
+        hostType: PLATFORM_TYPE_GITHUB,
+        matchHost: 'https://api.github.com',
+        token: 'defaultToken',
+      });
+      // specific host rule for using other token in different org
+      add({
+        hostType: PLATFORM_TYPE_GITHUB,
+        matchHost: 'https://api.github.com/repos/org-b/',
+        token: 'overwrittenToken',
+      } as any);
+      expect(
+        find({
+          hostType: PLATFORM_TYPE_GITHUB,
+          url: 'https://api.github.com/repos/org-b/someRepo/tags?per_page=100',
+        }).token
+      ).toEqual('overwrittenToken');
+    });
+    it('matches for several hostTypes when no hostType rule is configured', () => {
+      add({
+        matchHost: 'https://api.github.com',
+        token: 'abc',
+      });
+      expect(
+        find({
+          hostType: PLATFORM_TYPE_GITHUB,
+          url: 'https://api.github.com/repos/org-b/someRepo/tags?per_page=100',
+        }).token
+      ).toEqual('abc');
+      expect(
+        find({
+          hostType: 'github-releases',
+          url: 'https://api.github.com/repos/org-b/someRepo/tags?per_page=100',
+        }).token
+      ).toEqual('abc');
+    });
+    it('matches if hostType is configured and host rule is filtered with datasource', () => {
+      add({
+        hostType: PLATFORM_TYPE_GITHUB,
+        matchHost: 'https://api.github.com',
+        token: 'defaultToken',
+      });
+      add({
+        hostType: 'github-tags',
+        matchHost: 'https://api.github.com/repos/org-b/',
+        token: 'overwrittenToken',
+      } as any);
```

Comment:
```suggestion
      });
```

Cast shouldn't required